pool.go

```diff
@@ -61,3 +74,31 @@ func (p *ObjectPool) Put(obj interface{}) {
 	default:
 	}
 }
+
+// Init initializes the string slice pool.
+func (p *StringSlicePool) Init(alloc func() []string) {
+	p.alloc = alloc
+
+	for i := 0; i < cap(p.values); i++ {
+		p.values <- p.alloc()
+	}
+}
+
+// Get gets a string slice from the pool.
+func (p *StringSlicePool) Get() []string {
+	var v []string
+	select {
+	case v = <-p.values:
+	default:
+		v = p.alloc()
+	}
+	return v
+}
+
+// Put releases a string slice back to the pool.
+func (p *StringSlicePool) Put(obj []string) {
```

Comment:
Interestingly Put(obj interface{}) with a []string escapes to heap but not with *bytes.Buffer
That is why to save an alloc a specialization is done for []string

Reply:
Could you use a `sync.Pool` instead? We've been trying to move away from channel-based pools since these predate significant improvements in the runtime for `sync.Pool`.

You would have to actually use the pool directly instead of wrapping it, and storing `*[]string` slices (pointers _to_ slice header), and use it like:
```
buf := _stringPool.Get().(*[]string)

fpr {
	// do something - sanitize, etc...
	*buf = append(*buf, foobar)
}

*buf = (*buf)[:0]
_stringPool.Put(buf)
```

Note that if you cast `*[]string` back to `[]string` by doing `goodLookingBuffer := *buf` it can escape, because with Go you sometimes just can't have nice things.

Reply:
thanks for that hint. I used sync.Pool now and it works nicely. I only used it for key_gen.go only and ObjectPool is still being used by reporter.